Make minor improvements, add ToC (#11104)
galaxyproject/galaxy: doc/source/dev/writing_tests.md

          - **Yes**

            Tests that require a custom Galaxy with a very specific 
            configuration are called Galaxy integration tests and are located in
            ``test/integration``.  Checkout [Integration Tests](#integration)
            below for more information.

      - **Yes**

        The tests that exercise the Galaxy user interface and require

>         a functional Galaxy server all use Selenium to drive interaction

Reviewer: ```suggestion
        a functional Galaxy server use [Selenium](https://www.selenium.dev/) to drive interaction
```

## <a name="framework"></a> Tool Framework Tests

A great deal of the complexity and interface exposed to Galaxy plugin 
developers comes in the form of Galaxy tool wrapper definition files.
Likewise, a lot of the legacy behavior Galaxy needs to maintain is
maintained for older tool definitions. For this reason, a lot of Galaxy's
complex internals can just be tested by simply running a tool test.

> Obviously, Galaxy is much more complex than this, but a surprising amount

Reviewer: ```suggestion
Obviously Galaxy is much more complex than this, but a surprising amount
```

tests cannot be executed against external Galaxy servers (it wouldn't make
sense to, given these custom hooks during configuration of the server).
For these reasons, we bundle up Galaxy API tests for use in deployment

> testing of production setups; Galaxy API tests should be implemented whenever
> possible and integration tests implemented only when an API test is not

Reviewer: ```suggestion
testing of production setups. Therefore Galaxy API tests are generally preferred,
while integration tests should be implemented only when an API test is not
```
